components/proxy/src/main/java/com/hotels/styx/admin/handlers/MetricsHandler.java

```diff
@@ -39,6 +57,58 @@
      * @param cacheExpiration duration for which generated page content should be cached
      */
     public MetricsHandler(CodaHaleMetricRegistry metricRegistry, Optional<Duration> cacheExpiration) {
-        super(checkNotNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+        super(requireNonNull(metricRegistry.getMetricRegistry()), cacheExpiration, new MetricsModule(SECONDS, MILLISECONDS, DO_NOT_SHOW_SAMPLES));
+
+        this.metricRegistry = metricRegistry;
+    }
+
+    @Override
+    public Observable<HttpResponse> handle(HttpRequest request) {
+        String path = removeFinalSlash(request.path());
+
+        if (path.equals("/admin/metrics")) {
```

Comment:
Can't we find a way to do this avoiding this string duplication (/admin/metrics) ?  I think in this case it might be justified to design bearing in mind that we might also want to expose also specific origins... I mean, do it slightly more generic.
  
Also, the metrics are documented and displayed in the metrics endpoint  with dots. E.g., "requests.received" . I don't see why we are using slashes in the endpoint for specific metrics,  i would say it's an unnecessary and unwanted change.

Reply:
The idea was to use the URL path components to separate metric scopes. But not sure if that matters, really.

Reply:
I would say that in that case, we would also change the 'global metrics endpoint' and the docs to use those scope separators.

Reply:
Hi. Yes I’ve already left a comment about docs.

Could you elaborate your comment about ‘global metrics endpoint’?

Reply:
The '/admin/metrics' endpoint which contains all metrics with values such as: 
    "connections.openssl.session.accept" ,
    "connections.openssl.session.acceptGood"

In my opinion, if we supported metric scopes we wouldn't display the metrics in this endpoint as connections.openssl but  as 'connections/openssl' (if we had a scope named 'connections').   Why would the displayed metric name have to be transformed in order to query individual parameters?